packages/cli-platform-ios/src/tools/getBuildConfigurationFromXcScheme.ts

```diff
@@ -0,0 +1,43 @@
+import {CLIError} from '@react-native-community/cli-tools';
+import {XMLParser} from 'fast-xml-parser';
+import fs from 'fs';
+import path from 'path';
+
+const xmlParser = new XMLParser({ignoreAttributes: false});
+
+export function getBuildConfigurationFromXcScheme(
+  scheme: string,
+  configuration: string,
+  sourceDir: string,
+) {
+  try {
+    const xcProject = fs
+      .readdirSync(sourceDir)
+      .find((dir) => dir.includes('.xcodeproj'));
+
+    if (xcProject) {
+      const xmlScheme = fs.readFileSync(
+        path.join(
+          sourceDir,
+          xcProject,
+          'xcshareddata',
+          'xcschemes',
+          `${scheme}.xcscheme`,
+        ),
+        {
+          encoding: 'utf-8',
+        },
+      );
+
+      const {Scheme} = xmlParser.parse(xmlScheme);
```

Comment:
did you get a chance to asses how much extra time parsing this XML adds to a sample project?

Reply:
good idea, thanks - looks like on my machine the first read of the schema file takes ~5.5ms and every next try is around ~2.2ms - seems quick